libp2p.nimble

```diff
@@ -109,52 +99,23 @@ task examples_build, "Build the samples":
     buildSample("tutorial_3_protobuf", true)
     buildSample("tutorial_4_gossipsub", true)
     buildSample("tutorial_5_discovery", true)
-    # Nico doesn't work in 1.2
-    exec "nimble install -y nimpng@#HEAD" # this is to fix broken build on 1.7.3, remove it when nimpng version 0.3.2 or later is released
-    exec "nimble install -y nico"
     buildSample("tutorial_6_game", false, "--styleCheck:off")
 
 # pin system
 # while nimble lockfile
 # isn't available
 
-const PinFile = ".pinned"
-task pin, "Create a lockfile":
-  # pinner.nim was originally here
-  # but you can't read output from
-  # a command in a nimscript
-  exec "nim c -r tools/pinner.nim"
-
-import sequtils
-import os
-task install_pinned, "Reads the lockfile":
-  let toInstall = readFile(PinFile).splitWhitespace().mapIt((it.split(";", 1)[0], it.split(";", 1)[1]))
-  # [('packageName', 'packageFullUri')]
-
-  rmDir("nimbledeps")
-  mkDir("nimbledeps")
-  exec "nimble install -y " & toInstall.mapIt(it[1]).join(" ")
-
-  # Remove the automatically installed deps
-  # (inefficient you say?)
-  let nimblePkgs =
-    if system.dirExists("nimbledeps/pkgs"): "nimbledeps/pkgs"
-    else: "nimbledeps/pkgs2"
-  for dependency in listDirs(nimblePkgs):
-    let
-      fileName = dependency.extractFilename
-      fileContent = readFile(dependency & "/nimblemeta.json")
-      packageName = fileName.split('-')[0]
-
-    if toInstall.anyIt(
-        it[0] == packageName and
-        (
-          it[1].split('#')[^1] in fileContent or # nimble for nim 2.X
-          fileName.endsWith(it[1].split('#')[^1]) # nimble for nim 1.X
-        )
-      ) == false or
-      fileName.split('-')[^1].len < 20: # safegard for nimble for nim 1.X
-        rmDir(dependency)
-
-task unpin, "Restore global package use":
-  rmDir("nimbledeps")
+task test_slim, "Runs the (slimmed down) test suite":
+  `testnative Task`()
```

Comment:
Why is that change necessary? To keep some environment?

Reply:
To keep the params passed to nimble.